src/api/app/services/project_status_controller_service/attrib_values_finder.rb

```diff
@@ -0,0 +1,31 @@
+module ProjectStatusControllerService
+  class AttribValuesFinder
+    attr_accessor :packages, :name, :namespace
+
+    def initialize(packages, namespace, name)
+      @packages = packages
+      @name = name
+      @namespace = namespace
+    end
+
+    def self.call
+      raise NotImplementedError
```

Comment:
We should avoid using `NotImplementedError` whenever we want to document that subclasses need to implement this.

Interesting read on this: http://chrisstump.online/2016/03/23/stop-abusing-notimplementederror/

This is an example from our codebase: https://github.com/openSUSE/open-build-service/blob/70d920d52767236d4202b252be49a0320ff81d85/src/api/app/controllers/webui/request_controller.rb#L320-L322

I've done this recently when refactoring the request controller.

Reply:
We have many more examples in our application doing the opposite. So I would assume your solution rather an exception than the rule ;)

BTW  https://refactoring.guru/design-patterns/abstract-factory/ruby/example#lang-features

Even thou I agree that NotImplemented is probably not done for that, I don't like to the let the developer in the dark here.. so are you fine with a custom error class saying to implement it?

Reply:
I'm fine with a custom error class which says to implement the abstract method.

Reply:
The code I linked to in my previous comment should also be changed once we agreed on the solution.

Reply:
@dmarcoux could you please take a look and tell me if is that ok?